src/adapter/src/coord/message_handler.rs

```diff
@@ -581,11 +581,19 @@ impl Coordinator {
             match ready {
                 Deferred::Plan(mut ready) => {
                     ready.ctx.session_mut().grant_write_lock(write_lock_guard);
-                    // Write statements never need to track catalog
-                    // dependencies.
-                    let resolved_ids = ResolvedIds(BTreeSet::new());
-                    self.sequence_plan(ready.ctx, ready.plan, resolved_ids)
-                        .await;
+                    if ready
```

Comment:
Can this instead use or extend `PeekValidity`?

Reply:
Just updated to use this struct. 

Something that I just noticed is that `PeekValidity` does not include types or functions. I'm not really sure if that's an issue or not. I'm not sure what would happen if a type involved in a peek is deleted while that peek is off thread.

Reply:
I'm also not sure. Would probably panic.

Reply:
I'll open an issue.